docs/src/debugging.md

```diff
@@ -14,27 +14,61 @@ this data after any operation that ClimaCore performs.
 
 ## Example
 
+### Print `NaNs` when they are found
+
+In this example, we add a callback that simply prints `NaNs found` every
+instance when they are detected in a `ClimaCore` operation.
+
+To do this, we need two ingredients:
+
+First, we need to enable the callback system:
 ```@example
 import ClimaCore
-using ClimaCore: DataLayouts
 ClimaCore.DebugOnly.call_post_op_callback() = true
+```
+
+The line `ClimaCore.DebugOnly.call_post_op_callback() = true` means that at the
+end of every `ClimaCore` operation, the function
+`ClimaCore.DebugOnly.post_op_callback` is called. By default, this function does
+nothing. So, the second ingredient is to define a method:
+```@example
 function ClimaCore.DebugOnly.post_op_callback(result, args...; kwargs...)
-  if any(isnan, parent(data))
-    println("NaNs found!")
-  end
+    if any(isnan, parent(result))
+        println("NaNs found!")
+    end
 end
-
-FT = Float64;
-data = DataLayouts.VIJFH{FT}(Array{FT}, zeros; Nv=5, Nij=2, Nh=2)
-@. data = NaN
 ```
+If needed, `post_op_callback` can be specialized or behave differently in
+different cases, but here, it only checks if `NaN`s are in the given that.
 
 Note that, due to dispatch, `post_op_callback` will likely need a very general
 method signature, and using `post_op_callback
 (result::DataLayouts.VIJFH, args...; kwargs...)` above fails (on the CPU),
 because `post_op_callback` ends up getting called multiple times with different
 datalayouts.
 
+Now, let us put everything together and demonstrate a complete example:
+
+```@example
+import ClimaCore
+ClimaCore.DebugOnly.call_post_op_callback() = true
+function ClimaCore.DebugOnly.post_op_callback(result, args...; kwargs...)
+    if any(isnan, parent(result))
+        println("NaNs found!")
+    end
+end
+
+FT = Float64
+data = ClimaCore.DataLayouts.VIJFH{FT}(Array{FT}, zeros; Nv=5, Nij=2, Nh=2)
+@. data = NaN
```

Comment:
```suggestion
@. data = NaN
ClimaCore.DebugOnly.call_post_op_callback() = false # hide
```
If the docs preserves this session for later jobs (in `@example`), then `call_post_op_callback` will continue returning `true`, and could fail for other examples. (it looks like the docs run with a float, and not an array, which will fail due to no method matching `parent`).